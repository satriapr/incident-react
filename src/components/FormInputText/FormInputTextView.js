import { TextField } from '@mui/material';
import { Controller } from 'react-hook-form';
import PropTypes from 'prop-types';
import React from 'react';

// Common form input text. React-hook-form
const FormInputText = ({ name, control, label, isMultiline }) => {
  return (
    <Controller
      name={name}
      control={control}
      rules={{ required: `${label} is required` }}
      render={({ field: { onChange, value }, fieldState: { error } }) => (
        <TextField
          onChange={onChange}
          value={value || ''}
          label={label}
          error={!!error}
          helperText={error ? error.message : null}
          fullWidth
          variant="outlined"
          multiline={isMultiline}
          rows={isMultiline ? 4 : 0}
        />
      )}
    />
  );
};

export default FormInputText;

FormInputText.defaultProps = {
  name: '',
  control: {},
  label: '',
  isMultiline: false
};

FormInputText.propTypes = {
  name: PropTypes.string,
  control: PropTypes.objectOf(PropTypes.any),
  label: PropTypes.string,
  isMultiline: PropTypes.bool
};
