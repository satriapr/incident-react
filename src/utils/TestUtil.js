import { shallow } from 'enzyme';
import * as React from 'react';

const singleSnapTest = (wrapper, description) => {
  test(description, () => {
    expect(wrapper).toMatchSnapshot();
  });
};

const testSnapshots = (Component, configs) => {
  describe('snapshots', () => {
    configs.forEach((config) => {
      const { props, description, state } = config;
      const wrapper = shallow(<Component {...props} />);
      if (state) wrapper.setState(state);
      singleSnapTest(wrapper, description);
    });
  });
};

export { testSnapshots };
