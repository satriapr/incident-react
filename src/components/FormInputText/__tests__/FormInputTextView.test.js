import FormInputText from '../FormInputTextView';
import { testSnapshots } from '../../../utils/TestUtil';

describe('FormInputText Component', () => {
  testSnapshots(FormInputText, [
    {
      description: 'FormInputText props',
      props: {
        name: '',
        control: {},
        label: '',
        isMultiline: false
      }
    }
  ]);
});
