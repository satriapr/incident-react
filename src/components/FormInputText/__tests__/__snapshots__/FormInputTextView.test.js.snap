// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInputText Component snapshots FormInputText props 1`] = `ShallowWrapper {}`;
